ppocr/data/simple_dataset.py

```diff
@@ -124,18 +128,35 @@ def get_ext_data(self):
             label = substr[1]
             img_path = os.path.join(self.data_dir, file_name)
             data = {'img_path': img_path, 'label': label}
+            
             if not os.path.exists(img_path):
                 continue
-            with open(data['img_path'], 'rb') as f:
-                img = f.read()
+            # TODO (yiakwy) : cache img_path
+            if data['img_path'] in self.img_cache:
+                img = self.img_cache[data['img_path']]
+                # used by imgaug transform
                 data['image'] = img
-            data = transform(data, load_data_ops)
+                if (read_count + len(self.data_lines)) % 100 == 0:
+                    # pp data loader init a threads pool iterate over dataset
+                    print(f"[SimpleDataset] [{threading.current_thread().name}] fast loaded {read_count}th img {data['img_path']}. ext_data/ext_data_num={len(ext_data)}/{ext_data_num}")
+            else:
+                with open(data['img_path'], 'rb') as f:
+                    img = f.read()
+                    data['image'] = img
+                self.img_cache[data['img_path']] = img
+                if (read_count + len(self.data_lines)) % 100 == 0:
+                    print(f"[SimpleDataset] [{threading.current_thread().name}] {read_count}th img {data['img_path']} readed.")
+            if data['img_path'] in self.data_cache:
+                data = self.data_cache[data['img_path']]
+            else:
+                data = transform(data, load_data_ops)
+                self.data_cache[data['img_path']] = data
 
+            # read ith image
+            read_count += 1
+            
             if data is None:
                 continue
-            if 'polys' in data.keys():
-                if data['polys'].shape[1] != 4:
-                    continue
```

Comment:
need to handle this situation

Reply:
Hi, this is where we need discuss. 

This condition leaded to **endless loop** and discarded every annotation line in detection task with points more than 4.

I have checked annotation files we generated and official dataset from ctw1500 (800M) and ICDAR2015 (80M) datasets which are used for finetuning and evaluation: they both have annotation lines with more than 4 points.

The root cause is utility file can only process polygon with fixed points:

> assert len(points) == 4, "shape of points must be 4*2"

Hence we deleted and training goes smoothly again.